mirrord/agent/src/main.rs

```diff
@@ -464,6 +468,69 @@ async fn start_agent() -> Result<()> {
     Ok(())
 }
 
+async fn create_iptable_chain() -> Result<String> {
+    let ipt = iptables::new(false).unwrap();
+    let formatter = IPTableFormatter::detect(&ipt)?;
+
+    SafeIpTables::create_chain(&ipt, &formatter)
+}
+
+async fn clear_iptable_chain(chain_name: String) -> Result<()> {
+    let ipt = iptables::new(false).unwrap();
+    let formatter = IPTableFormatter::detect(&ipt)?;
+
+    SafeIpTables::remove_chain(&ipt, &formatter, &chain_name)
+}
+
+fn spawn_child_agent() -> Result<()> {
+    use std::process::{Command, Stdio};
+
+    let command_args = std::env::args().collect::<Vec<_>>();
+
+    let mut child_agent = Command::new(&command_args[0])
+        .args(&command_args[1..])
+        .stdin(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .spawn()?;
+
+    let _ = child_agent.wait();
+
+    Ok(())
+}
+
+async fn start_iptable_guard() -> Result<()> {
+    debug!("start_iptable_guard -> Initializing iptable-guard.");
```

Comment:
imo the chain should be created in the child and the parent only cleans (doesn't create anything)
this will guard us from cases where the guard itself gets killed unexpectedly during creation